bootable-jar/WFCORE-5184_Check_Boot.adoc

```diff
@@ -0,0 +1,100 @@
+= Bootable JAR, check server and deployment state at boot time
+:author:           Jean-Francois Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This feature is to enforce that a Bootable JAR that successfully booted is in a valid state (server and deployment started without error).
+During boot, when an invalid state is encountered, the Bootable JAR process is aborted.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5184[WFCORE-5184]
+
+=== Related Issues
+
+* 
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* Bootable JAR runtime
+
+=== Other Interested Projects
+
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin/[Bootable JAR Maven plugin]
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [ ] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [ ] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+The JAR is auto-checking the server and deployment states at boot time in order to abort the process in case of errors.
+
+=== Hard Requirements
+
+* The boot checks are enabled by default. This is a change of behavior that needs to be documented.
+* Introduce `--check-boot=[true|false]` in order to disable the checks.
+* Prior to check for errors, the Bootable JAR waits for the server to reach the following states:
+** In case the server has been started with a CLI script to execute, the server running mode is expected to reach the NORMAL mode.
+** The server state is expected to be RUNNING. 
+* This new feature doesn't introduce a timeout to bound the server start time (that can be very long for large and complex application).
+* In case the server never reaches these states, the Bootable JAR doesn't take any decision and waits until the server is stopped (by any mean).
+* When the server is running the following checks are operated:
+** If boot errors are detected, the process abort
+** If a deployment exists and has a FAILED status, the process is aborted.
+* If no invalid conditions are met, the bootable JAR advertises that it is up and running.
+* If an invalid condition is met, the bootable JAR advertises the problem and exits (error code == 1).
+* It could happen that the server is aborting (Fatal server error), in this case, no action is taken by the Bootable JAR. 
+The server exits automatically (as done today).
+* Once the server has booted successfully, no more check is operated.
```

Comment:
I'm not sure I follow this. Do you mean it's extracted as normal?

Somewhat along these lines if the boot fails should the extracted server be deleted/cleaned up?

Reply:
I added a note to specify that the nominal cleanup is applied when the boot is aborted (actually System.exit(1) is called by the bootable JAR runtime so shutdown hook is called). 
The WIP btranch is there BTW: https://github.com/wildfly/wildfly-core/compare/master...jfdenise:WFCORE-5184

Reply:
I also added a note that says that the deployment status is not monitored once the server has booted. This is what I meant